moveit_core/macros/include/moveit/macros/declare_ptr.h

```diff
@@ -52,4 +52,19 @@
   typedef boost::shared_ptr<Type> Name##Ptr;          \
   typedef boost::shared_ptr<const Type> Name##ConstPtr;
 
+/**
+ * \def MOVEIT_DELCARE_PTR_MEMBER
+ * Macro that given a Type declares the following types:
+ * - Ptr      = shared_ptr<${Type}>
+ * - ConstPtr = shared_ptr<const ${Type}>
+ *
+ * This macro is intended for declaring the pointer typedefs as members of a
+ * class template. In other situations, MOVEIT_CLASS_FORWARD and
+ * MOVEIT_DECLARE_PTR should be preferred.
+ */
+
+#define MOVEIT_DECLARE_PTR_MEMBER(Type)         \
+  typedef boost::shared_ptr<Type> Ptr;          \
+  typedef boost::shared_ptr<const Type> ConstPtr;
+
```

Comment:
Since C++11, `MOVEIT_DECLARE_PTR(, Type)` is also legal (note the empty first argument), but pre-11 it is undefined what the preprocessor does with that. It's also not the prettiest to look at like, so that is why I added this variant on `MOVEIT_DECLARE_PTR`.